R/utilities-writing-report.R

```diff
@@ -709,3 +709,53 @@ updateTableNumbers <- function(fileContent, pattern = "Table:", replacement = "T
   return(updateFigureNumbers(fileContent, pattern, replacement, anchorId))
 }
 
+#' @title copyReport
+#' @description Copy markdown report and its figures (using their paths)
+#' @param from path of initial .md file to copy
+#' @param to path of destination .md file to be copied
+#' @param keep logical defining if initial .md file and figures are kept
+#' @export
+copyReport <- function(from, to, keep = FALSE) {
+  validateIsFileExtension(from, "md")
+  validateIsFileExtension(to, "md")
+  validateFileExists(from)
+  validateIsLogical(keep)
+  # If from is 
+  if(from == to){
+    return(invisible())
+  }
+  # Get directories of reports
+  fromFolder <- dirname(from)
+  toFolder <- dirname(to)
```

Comment:
is it possible that `from != to` but `fromFolder = toFolder ` ?
E.g. cann the function be called with like this: `copyReport("C:/temp/report1", "C:/temp/report2")`?

IF yes: we need additional check here, if `fromFolder `and `toFolder  `are equal. And if yes: no need to copy the images subfolder - only the report itself (md) must be copied

Reply:
Good point, I'll return without copying if folders are same